libs/hooks/open-telemetry/src/lib/traces/tracing-hook.ts

```diff
@@ -1,43 +1,98 @@
 
-export type TracingHookOptions = OpenTelemetryHookOptions;
+/**
+ * A hook that logs evaluation events to OpenTelemetry using an EventLogger.
+ * This is useful for exporting evaluation events to a backend that supports
+ * OpenTelemetry events.
+ */
+export class LogEventTracingHook extends OpenTelemetryHook implements BaseHook {
+  protected name = LogEventTracingHook.name;
+  private eventLogger: EventLogger;
+
+  constructor(options?: OpenTelemetryHookOptions, logger?: Logger) {
+    super(options, logger);
+    this.eventLogger = events.getEventLogger('feature-flag'); // TODO: Which name? User configurable?
+  }
+
+  finally(hookContext: Readonly<HookContext>, evaluationDetails: EvaluationDetails<FlagValue>) {
+    this.eventLogger.emit(this.toEvaluationEvent(hookContext, evaluationDetails));
+  }
+
+  error(_: HookContext, err: Error) {
+    trace.getActiveSpan()?.recordException(err);
+  }
+}
 
 /**
- * A hook that adds conventionally-compliant span events to feature flag evaluations.
- *
- * See {@link https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/feature-flags/}
+ * A hook that adds evaluation events to the current active span.
+ * This is useful for associating evaluation events with a trace.
+ * If there is no active span, the event is not logged.
+ * Span events are being deprecated in favor of using log events.
  */
-export class TracingHook extends OpenTelemetryHook implements Hook {
-  protected name = TracingHook.name;
+export class SpanEventTracingHook extends OpenTelemetryHook implements BaseHook {
+  protected name = SpanEventTracingHook.name;
 
-  constructor(options?: TracingHookOptions, logger?: Logger) {
+  constructor(options?: OpenTelemetryHookOptions, logger?: Logger) {
     super(options, logger);
   }
 
-  after(hookContext: HookContext, evaluationDetails: EvaluationDetails<FlagValue>) {
+  finally(hookContext: Readonly<HookContext>, evaluationDetails: EvaluationDetails<FlagValue>) {
     const currentTrace = trace.getActiveSpan();
-    if (currentTrace) {
-      let variant = evaluationDetails.variant;
-
-      if (!variant) {
-        if (typeof evaluationDetails.value === 'string') {
-          variant = evaluationDetails.value;
-        } else {
-          variant = JSON.stringify(evaluationDetails.value);
-        }
-      }
-
-      currentTrace.addEvent(FEATURE_FLAG, {
-        [KEY_ATTR]: hookContext.flagKey,
-        [PROVIDER_NAME_ATTR]: hookContext.providerMetadata.name,
-        [VARIANT_ATTR]: variant,
-        ...this.safeAttributeMapper(evaluationDetails.flagMetadata),
-      });
+    if (!currentTrace) {
+      return;
     }
+
+    const { name, attributes } = this.toEvaluationEvent(hookContext, evaluationDetails);
+
+    currentTrace.addEvent(name, attributes);
+  }
+
+  error(_: HookContext, err: Error) {
+    trace.getActiveSpan()?.recordException(err);
+  }
+}
+
+const tracer = trace.getTracer('span-attributes-tracing-hook');
```

Comment:
This should work the same way as described [here](https://github.com/open-feature/js-sdk-contrib/pull/1372/files#r2392408426).